modules/openid_auth/README.yml

```diff
@@ -0,0 +1,145 @@
+openapi: '3.0.0'
+info:
+  version: 0.0.1
+  title: Validation
+  description: |
+    Oauth Validation for internal 3rd Party Services
+
+    ## Background
+
+    
+
+    ## Design
+
+    ### Authorization
+
+    API requests are authorized using a Bearer token issued through an OpenID
+    Connect service to allow third-party applications. The token should be
+    submitted as an `Authorization` header in the form `Bearer <token>`.
+
+    ### Veteran Status Request
+
+    Allows a third-party application to request the validated token and VA Identifier ICN for an individual:
+
+    1. Client Request: GET https://api.vets.gov/internal/auth/v0/validation
+       * Provide the Bearer token as a header: `Authorization: Bearer <token>`
+
+    2. Status Response: A JSON API object with the individual's validated token and VA identifier ICN
+
+    ## Reference
+
+    Raw Open API Spec: http://dev-api.va.gov/internal/auth/docs/v0/validation
+
+  termsOfService: ''
+  contact:
+    name: VA.gov
+tags:
+  - name: validated_token
+    description: Oauth Validation for internal 3rd Party Services
+servers:
+  - url: dev-api.va.gov/internal/auth/{version}
+    description: VA.gov API development environment
+    variables:
+      version:
+        default: v0
+  - url: staging-api.va.gov/internal/auth/{version}
+    description: VA.gov API staging environment
+    variables:
+      version:
+        default: v0
+  - url: api.va.gov/internal/auth/{version}
+    description: VA.gov API production environment
+    variables:
+      version:
+        default: v0
+paths:
+  /status:
+    get:
+      tags:
+        - validated_token
+      summary: Get confirmation of a validated token and VA identifier ICN
+      operationId: getValidatedToken
+      security:
+        - bearer_token: []
+      responses:
+        '200':
+          description: Validated token successfully retrieved
+          content:
+            application/json:
+              schema:
+                required:
+                  - data
+                properties:
+                  data:
+                    $ref: '#/components/schemas/ValidatedToken'
+        '403':
+          description: Provided Access Token Does not have the required scopes to access.
+        '502':
```

Comment:
We don't look up from eMIS, I don't think this error response is relevant for this API.

Reply:
I seee have a catch-all that returns a 502 in the controller but that means "Bad Gateway". We're not proxying through to any other service at that point so should instead just be a 500 error meaning we couldn't assemble the response

Reply:
Created a 502 description to say `Token failed to be validated.` per your request.